src/GoTrueClient.ts

```diff
@@ -884,17 +889,16 @@ export default class GoTrueClient {
     }
   }
 
-  private async _callRefreshToken(refresh_token = this.currentSession?.refresh_token) {
-    try {
-      // refreshing is already in progress
-      if (this.refreshingDeferred) {
-        return await this.refreshingDeferred.promise
-      }
+  private async _callRefreshToken(
+    refresh_token = this.currentSession?.refresh_token
+  ): Promise<CallRefreshTokenResult> {
+    // refreshing is already in progress
+    if (this.refreshingDeferred) {
+      return this.refreshingDeferred.promise
```

Comment:
Moved that out of the try block and removed the `await`. Otherwise a rejection would fall into `catch` and `finally` multiple times. As `_callRefreshToken` throws non `AuthErorr`'s that should be the expected behavior.